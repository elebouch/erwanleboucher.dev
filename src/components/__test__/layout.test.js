import React from "react"
import { render } from "@testing-library/react"
import "jest-styled-components"

import Layout from "../layout"

describe("Layout", () => {
  it("renders correctly", () => {
    const { container } = render(<Layout />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
