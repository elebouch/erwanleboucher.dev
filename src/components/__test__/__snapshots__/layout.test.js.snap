// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
.c8 {
  position: relative;
  display: inline-block;
  width: 40px;
  height: 20px;
  background-color: var(--white);
  -webkit-transition: background-color 0.3s ease-in;
  transition: background-color 0.3s ease-in;
  border-radius: 20px;
  border: 2px solid var(--primary);
  cursor: pointer;
}

.c8::after {
  content: "";
  position: absolute;
  width: 18px;
  height: 18px;
  border-radius: 50%;
  background-color: #fad21c;
  top: 1px;
  left: 2px;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c6 {
  display: none;
}

.c6:checked + .c7::after {
  background-color: transparent;
  box-shadow: 5px 5px 0 0 var(--dark);
  left: 13px;
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

.c6:checked + .c7 {
  background-color: #000;
  border: 2px solid var(--dark);
  -webkit-transition: background-color 0.3s ease-in;
  transition: background-color 0.3s ease-in;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 10px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  margin: 10px 0px;
  color: var(--dark);
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: "Archivo Black",sans-serif;
  font-size: 36px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 10px;
}

.c4 {
  margin-right: 10px;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: "Archivo Black",sans-serif;
  font-weight: bold;
  color: var(--dark);
}

.c0 {
  max-width: 800px;
  margin: 40px auto;
}

.c9 {
  height: 100%;
  width: 100%;
  margin: 20px 10px;
}

.c1 {
  position: absolute;
  top: -40px;
  left: 0;
  background: #000000;
  color: white;
  padding: 8px;
  z-index: 100;
}

.c1:focus {
  top: 0;
}

@media (max-width:800px) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    text-align: center;
  }
}

@media (max-width:800px) {
  .c5 {
    margin-top: 10px;
    margin-right: 0px;
  }
}

@media (max-width:800px) {
  .c0 {
    margin: 0;
  }
}

@media (max-width:800px) {
  .c9 {
    margin: 0;
  }
}

<div
  class="c0"
>
  <a
    class="c1"
    href="#maincontent"
  >
    Skip to main
  </a>
  <header>
    <div
      class="c2"
    >
      <a
        class="c3 c4"
        href="/"
      >
        test
      </a>
      <div
        class="c5"
      >
        <a
          class="c4"
          href="/"
        >
          About
        </a>
        <a
          class="c4"
          href="/journey"
        >
          Journey
        </a>
        <a
          class="c4"
          href="/projects"
        >
          Projects
        </a>
        <a
          class="c4"
          href="/blog"
        >
          Blog
        </a>
        <input
          class="c6"
          data-testid="theme-checkbox"
          id="switch"
          type="checkbox"
        />
        <label
          class="c7 c8"
          for="switch"
        />
      </div>
    </div>
  </header>
  <div
    class="c9"
    id="maincontent"
  />
</div>
`;
